docs/core/store.md

```diff
@@ -0,0 +1,39 @@
+<!--
+order: 3
+-->
+
+# Store and Database
+
+Learn about the store and database concept on Ethermint. {synopsis}
```

Comment:
should we add links to learn more about the cosmos sdk modules?
https://docs.cosmos.network/master/core/store.html